libstorage/ZdbStorage.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2019 fisco-dev contributors.
+ */
+/** @file ZdbStorage.cpp
+ *  @author darrenyin
+ *  @date 2019-04-24
+ */
+
+#include "ZdbStorage.h"
+#include "Common.h"
+#include "StorageException.h"
+#include "Table.h"
+#include <libdevcore/easylog.h>
+
+
+using namespace dev;
+using namespace dev::storage;
+
+ZdbStorage::ZdbStorage() {}
+
+Entries::Ptr ZdbStorage::select(
+    h256 hash, int num, TableInfo::Ptr tableInfo, const std::string& key, Condition::Ptr condition)
+{
+    Json::Value responseJson;
+    int iRet = m_oSqlBasicAcc.Select(hash, num, tableInfo->name, key, condition, responseJson);
+    if (iRet < 0)
+    {
+        LOG(ERROR) << "Remote database return error:" << iRet;
+        throw StorageException(
+            -1, "Remote database return error:" + boost::lexical_cast<std::string>(iRet));
+    }
+
+
+    LOG(DEBUG) << "select resp:" << responseJson.toStyledString();
+    std::vector<std::string> columns;
+    for (Json::ArrayIndex i = 0; i < responseJson["result"]["columns"].size(); ++i)
+    {
+        std::string fieldName = responseJson["result"]["columns"].get(i, "").asString();
+        columns.push_back(fieldName);

```

Comment:
too many copy here
1,asString 2, copy to fieldName 3,copy to columns

Reply:
fixed